config_test.go

```diff
@@ -64,4 +64,171 @@ func TestConfig(t *testing.T) {
 			t.Fatal("cannot work set/get with tag")
 		}
 	})
+
+	t.Run("llc tags", func(t *testing.T) {
+		t.Run("cache control", func(t *testing.T) {
+			t.Run("with single transaction", func(t *testing.T) {
+				t.Run("explicit no cache control", func(t *testing.T) {
+
+					tagName := "cache_control_no_lock"
+
+					t.Run("should have false cache control config", func(t *testing.T) {
+						tagCacheControlConfig := (*cfg.LLC.Tags)[tagName].CacheControl
+						Equalf(t, *tagCacheControlConfig.OptimisticLock, false, "should be false")
+						Equalf(t, *tagCacheControlConfig.PessimisticLock, false, "should be false")
+					})
+					t.Run("options should have false values", func(t *testing.T) {
+						tagOption := cache.lastLevelCache.opt.tagOpt[tagName]
+						Equalf(t, *tagOption.optimisticLock, false, "should be false")
+						Equalf(t, *tagOption.pessimisticLock, false, "should be false")
+					})
+					t.Run("should retrieve last created value", func(t *testing.T) {
+						key := fmt.Sprintf("key_%d", time.Now().UnixNano())
+						var result int
+						expect := 2
+
+						tx, err := cache.Begin()
+						NoError(t, err)
+
+						NoError(t, tx.CreateWithTag(tagName, key, Int(1)))
+						NoError(t, tx.CreateWithTag(tagName, key, Int(expect)))
+						NoError(t, tx.FindWithTag(tagName, key, IntPtr(&result)))
+
+						Equalf(t, result, expect, "should retrieve last created value")
+					})
+				})
+
+				t.Run("explicit lock", func(t *testing.T) {
+
+					tagName := "cache_control_lock"
+
+					t.Run("config should have true cache control config", func(t *testing.T) {
+						tagCacheControlConfig := (*cfg.LLC.Tags)[tagName].CacheControl
+						Equalf(t, *tagCacheControlConfig.OptimisticLock, true, "should be true")
+						Equalf(t, *tagCacheControlConfig.PessimisticLock, true, "should be true")
+					})
+					t.Run("options should have true values", func(t *testing.T) {
+						tagOption := cache.lastLevelCache.opt.tagOpt[tagName]
+						Equalf(t, *tagOption.optimisticLock, true, "should be true")
+						Equalf(t, *tagOption.pessimisticLock, true, "should be true")
+					})
+					t.Run("should retrieve last created value", func(t *testing.T) {
+						key := fmt.Sprintf("key_%d", time.Now().UnixNano())
+						var result int
+						expect := 2
+
+						tx, err := cache.Begin()
+						NoError(t, err)
+						NoError(t, tx.CreateWithTag(tagName, key, Int(1)))
+						NoError(t, tx.CreateWithTag(tagName, key, Int(expect)))
+						NoError(t, tx.FindWithTag(tagName, key, IntPtr(&result)))
+
+						Equalf(t, result, expect, "should retrieve last created value")
+					})
+				})
+
+				t.Run("implicit cache control", func(t *testing.T) {
+
+					tagName := "cache_control_implicit"
+
+					t.Run("config should have nil values", func(t *testing.T) {
+						tagCacheControlConfig := (*cfg.LLC.Tags)[tagName].CacheControl
+						IsNilf(t, tagCacheControlConfig, "should be nil")
```

Comment:
I would think better to use `if tagCacheControlConfig == nil { t.Fatalf("....") }` because `IsNil(f)` is used only here ( and the below statements ) .

Reply:
OK, that's YAGNI.
I'm gonna fix it, thank you.